tests/test_api/test_validation.py

```diff
@@ -84,75 +84,215 @@ def test_check_dem_coverage():
+                    'S1_136231_IW2_20200617T022313_VV_5D11-BURST',
+                    'S1_136233_IW2_20200617T123213_VV_5E13-BURST'
+                ]
+            }
         },
         {
-            'name': 'S1_136231_IW2_20200616T022313_HH_5D11-BURST'
+            'job_parameters': {
+                'reference': [
+                    'S1_136232_IW2_20200604T022312_VV_7C85-BURST',
+                    'S1_136231_IW2_20200604T123455_VV_ABC5-BURST'
+                ],
+                'secondary': [
+                    'S1_136231_IW2_20200617T022313_VV_5D11-BURST',
+                    'S1_136233_IW2_20200617T123213_VV_5E13-BURST'
+                ]
+            }
         }
     ]
-    unsupported_polarizations = [
+    invalid_jobs_duplicate = [
         {
-            'name': 'S1_136231_IW2_20200604T022312_VH_7C85-BURST'
+            'job_parameters': {
+                'reference': [
+                    'S1_136231_IW2_20200604T022312_VV_7C85-BURST',
+                    'S1_136231_IW2_20200604T123455_VV_ABC5-BURST'
+                ],
+                'secondary': [
+                    'S1_136231_IW2_20200617T022313_VV_5D11-BURST',
+                    'S1_136231_IW2_20200617T123213_VV_5E13-BURST'
+                ]
+            }
         },
         {
-            'name': 'S1_136231_IW2_20200616T022313_VH_5D11-BURST'
+            'job_parameters': {
+                'reference': [
+                    'S1_136231_IW2_20200604T022312_VV_7C85-BURST',
+                    'S1_136231_IW2_20200604T123455_VV_ABC5-BURST',
+                    'S1_136232_IW2_20200604T125455_VV_ABC6-BURST',
+                ],
+                'secondary': [
+                    'S1_136231_IW2_20200617T022313_VV_5D11-BURST',
+                    'S1_136231_IW2_20200617T123213_VV_5E13-BURST',
+                    'S1_136232_IW2_20200604T123475_VV_ABC7-BURST',
+                ]
+            }
         }
     ]
+    for valid_job in valid_jobs:
+        validation.check_same_burst_ids(valid_job, {})
+    for invalid_job in invalid_jobs_not_matching:
+        with raises(validation.GranuleValidationError, match=r'.*burst IDs with no matching pairs*'):
+            validation.check_same_burst_ids(invalid_job, {})
+    for invalid_job in invalid_jobs_duplicate:
+        with raises(validation.GranuleValidationError, match=r'.*more than 1 pair with the same burst ID*'):
+            validation.check_same_burst_ids(invalid_job, {})
+
 
-    validation.check_valid_polarizations(valid_case)
-    with raises(validation.GranuleValidationError, match=r'.*do not have the same polarization.*'):
-        validation.check_valid_polarizations(different_polarizations)
-    with raises(validation.GranuleValidationError, match=r'.*Only VV and HH.*'):
-        validation.check_valid_polarizations(unsupported_polarizations)
+def test_check_valid_polarizations():
+    valid_cases = [
+        [
+            {
+                'name': 'S1_136231_IW2_20200604T022312_VV_7C85-BURST'
```

Comment:
If my suggestion at https://github.com/ASFHyP3/hyp3/pull/2386/files#r1725857878 is accepted, then `name` will need to be replaced by `reference` and `secondary`, like you did above.